kmod/src/server.c

```diff
@@ -3536,6 +3550,15 @@ static int server_farewell(struct super_block *sb,
 	list_add_tail(&fw->entry, &server->farewell_requests);
 	spin_unlock(&server->farewell_lock);
 
+	/* Tear down client lock server state and set that we recieved farewell
+	 * to ensure that we do not race between client and server trying to process
+	 * lock recovery at the same time (race). We also want to mark that the recovery
+	 * finished so that if client's try to send stuff later; the server doesnt care.
+	 */
+	sci->received_farewell = true;
+	scoutfs_lock_server_farewell(sb, rid);
```

Comment:
_lock_server_farewell() can return an error.   Removing the client's locks can unblock other locks that were pending.   It can return an error if it can't advance the state of those blocked locks (maybe by sending messages to other clients), which is really bad and means that the lock server is stuck.   So if it gets an error free the fw we allocated and return the error, that'll shut down the server.